### PR TITLE
Remove obsolete parser code

### PR DESCRIPTION
**Pull request by tlively**
Remove `SExpressionParser`, `SExpressionWasmBuilder`, and `cashew::Parser`.
Simplify gen-s-parser.py. Remove the --new-wat-parser and
--deprecated-wat-parser flags.

### CODE REVIEW

**Comment by tlively** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6607?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#6607** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6607?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6606** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6606?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tlively and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/emscripten-optimizer/parser.h [hunk 1]
```diff
@@ -186,988 +184,6 @@ extern std::vector<OperatorClass> operatorClasses;
 extern bool isIdentInit(char x);
 extern bool isIdentPart(char x);
 
-// parser
```
Comment (kripken) [inline src/emscripten-optimizer/parser.h:189]: This parses JS, which I guess could be a separate PR. But maybe not worth it if it's just a safe removal.

**Review by kripken — APPROVED** [review]
